docs/src/examples.md

```diff
@@ -0,0 +1,139 @@
+Examples
+========
+
+The K8sClusterManager is intended to be used inside a pod running on a Kubernetes cluster.
+
+## Launching an interactive session
+
+The following manifest will create a [Kubernetes job](https://kubernetes.io/docs/concepts/workloads/controllers/job/)
+named "interactive-session". This job will spawn a pod (see `spec.template.spec`) which will
+run an interactive Julia session with the latest release of K8sClusterManagers.jl installed.
+Be sure to create the required [service account and associated permissions](../patterns/#required-permissions)
+before proceeding.
+
+````@eval
+using Markdown
+Markdown.parse("""
+```yaml
+$(read("interactive-session.yaml", String))
+```
+""")
+````
+
+To start the job and attach to the interactive Julia session you can run the following:
+
+```sh
+# Executed from the K8sClusterManager.jl root directory
+kubectl apply -f docs/src/interactive-session.yaml
+
+# Determine the generated pod name from the job
+manager_pod=$(kubectl get pods -l job-name=interactive-session --output=jsonpath='{.items[*].metadata.name}')
+echo $manager_pod
+
+# Attach to the interactive Julia session running in the pod.
+# Note: You may need to wait for K8sClusterManagers.jl to finish installing
+kubectl attach -it pod/${manager_pod?}
+```
+
+### Launching workers
+
+Once you've attached to the interactive session you can use [`K8sClusterManager`](@ref) to
+spawn k8s workers. For our example we'll be using a small amount of CPU/Memory to ensure
+workers can be spawned even on clusters with limited resources:
+
+```julia
+julia> using Distributed, K8sClusterManagers
+
+julia> addprocs(K8sClusterManager(3, cpu=0.2, memory="300Mi", pending_timeout=30))
+[ Info: interactive-example-sp28h-worker-7dvpt is up
+[ Info: interactive-example-sp28h-worker-vvrwm is up
+[ Info: interactive-example-sp28h-worker-bczm5 is up
+3-element Vector{Int64}:
+ 2
+ 3
+ 4
+
+julia> pmap(x -> myid(), 1:nworkers())  # Each worker reports its worker ID
+3-element Vector{Int64}:
+ 3
+ 4
+ 2
+```
+
+### Pending workers
+
+A common issue when spawning workers is not having enough resources available to start the
+workers. Worker pods which cannot be started will be stuck in the "Pending" phase and will
+wait until resources become available. Since it is not known how long a worker may be stuck
+in the "Pending" phase the [`K8sClusterManager`](@ref) includes the `pending_timeout`
+keyword which specifies how long you are willing to wait for pending workers. Once this
+timeout has been reached the manager will continue with the subset of workers which have
+reported in.
```

Comment:
```suggestion
A worker created via `addprocs` may not necessarily be available right away, as K8s must schedule
the worker's underlying *Pod* to an actual *Node* before the corresponding Julia process can start.
If K8s can't schedule the worker's *Pod* to a *Node* right away (e.g. there might not be enough resources 
available), then that *Pod* will sit around in the "Pending" phase until it can be scheduled. Since a worker *Pod* 
may be stuck in this "Pending" phase for an indefinite amount of time, [`K8sClusterManager`](@ref) includes 
the `pending_timeout` keyword that may used to specify how long you are willing to wait for pending workers. 
Once this timeout has been reached, the manager will continue with the subset of workers which have reported in
and drop any workers that are stuck in the "Pending" phase.
```

simple edit for clarity; could you check "drop any workers that are stuck" here to make sure the statement is accurate? if not, can replace with a more accurate statement

Reply:
Overall quite accurate. I'll just change "drop" to "delete" to match the K8s verb and make a minor tweak to the example